core/Enclave/enclave_hsm.cpp

```diff
@@ -246,28 +382,30 @@ sgx_status_t enclave_create_aes_key(uint8_t *cmk_blob, uint32_t cmk_blob_size, u
 
 /*
  * struct cipherblob {
- *    OUT uint8_t ciphertext[SGX_AES_KEY_SIZE];
+ *    OUT uint8_t ciphertext[KEY_SIZE];
  *    OUT uint8_t iv[EH_AES_GCM_IV_SIZE];   // 12B
  *    OUT uint8_t mac[EH_AES_GCM_MAC_SIZE]; // 16B
  * }
  */
+
 sgx_status_t enclave_aes_encrypt(const uint8_t *aad, size_t aad_len,
-                             const uint8_t *cmk_blob, size_t cmk_blob_size,
-                             const uint8_t *plaintext, size_t plaintext_len,
-                             uint8_t *cipherblob, size_t cipherblob_len)
+                                 const uint8_t *cmk_blob, size_t cmk_blob_size,
+                                 const uint8_t *plaintext, size_t plaintext_len,
+                                 uint8_t *cipherblob, size_t cipherblob_len, ehsm_keyspec_t keyspec)
 {
     sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+    uint32_t key_size = sgx_get_key_size(keyspec);
 
     if (cmk_blob == NULL)
         return SGX_ERROR_INVALID_PARAMETER;
 
-    uint32_t real_cmk_blob_size = sgx_calc_gcm_data_size(0, SGX_AES_KEY_SIZE);
+    uint32_t real_cmk_blob_size = sgx_calc_gcm_data_size(0, key_size);
     if (UINT32_MAX == real_cmk_blob_size || cmk_blob_size < real_cmk_blob_size)
         return SGX_ERROR_INVALID_PARAMETER;
 
     uint32_t enc_key_size = sgx_get_gcm_ciphertext_size((sgx_aes_gcm_data_ex_t *)cmk_blob);
-    if (enc_key_size == UINT32_MAX || enc_key_size != sizeof(sgx_key_128bit_t)) {
-        printf("enc_key_size:%d is not expected: %lu.\n", enc_key_size, sizeof(sgx_key_128bit_t));
+    if (enc_key_size == UINT32_MAX || enc_key_size != sizeof(uint8_t[key_size])) {
```

Comment:
sizeof(uint8_t[key_size]  -> key_size

Reply:
It should really be written like this. I'll change.